addons/ondevice-controls/src/types/Date.tsx

```diff
@@ -23,82 +23,66 @@ const Label = styled.Text(({ theme }: any) => ({
   color: theme.labelColor || 'black',
 }));
 
-// TODO seconds support
-class DateType extends PureComponent<
-  DateProps,
-  { isTimeVisible: boolean; isDateVisible: boolean }
-> {
-  static defaultProps = {
-    knob: {},
-    onChange: (value) => value,
-  };
-
-  static serialize = (value) => String(value);
-
-  static deserialize = (value) => parseFloat(value);
-
-  constructor(props) {
-    super(props);
-    this.state = {
-      isDateVisible: false,
-      isTimeVisible: false,
-    };
-  }
+const DateType = ({ onChange, arg: { name, value } }: DateProps) => {
+  const [isDateVisible, setIsDateVisible] = useState(false);
```

Comment:
This split to two different boolean variables (`isDateVisible` and `isTimeVisible`) pre-dates your code, but I'm tempted to suggest a small refactor here: https://github.com/storybookjs/react-native/pull/225

It's by no means a priority or a blocker, but thought you might be able to take a quick peek now that this code is fresh in the minds of both of us.

Reply:
accepted your PR, seems like a great idea to me. Thanks again!